src/NuGet.Core/NuGet.Protocol/Utility/DedicatedAsynchronousProcessingThread.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Protocol
+{
+    /// <summary>
+    /// This class represents a dedicated asynchronous task processing thread.
+    /// Uses a queue to execute all the tasks added through the invocation of <see cref="Enqueue(Func{Task})"/>.
+    /// The tasks queued here cannot be awaited (think Task.Run, rather than Task.WhenAny/WhenAll).
+    /// This implementation is internal on purpose as this is specifically tailed to the plugin V2 use-case.
+    /// </summary>
+    internal class DedicatedAsynchronousProcessingThread : IDisposable
+    {
+        private Task _processingThread;
+        private bool _isDisposed;
+        private bool _isClosed;
+        private readonly int _pollingDelayInMilliseconds;
+        private readonly ConcurrentQueue<Func<Task>> _taskQueue = new ConcurrentQueue<Func<Task>>();
+
+        /// <summary>
+        /// DedicatedAsync processing thread.
+        /// </summary>
+        /// <param name="pollingDelayInMilliseconds">The await delay when there are no tasks in the queue.</param>
+        public DedicatedAsynchronousProcessingThread(int pollingDelayInMilliseconds = 50)
```

Comment:
Consider using `TimeSpan`.  Then you don't need to explain (`InMilliseconds`) or hard-code units.

`TimeSpan pollingDelay`

Later:
```C#
await Task.Delay((int)_pollingDelay.TotalMilliseconds);
```

That said, I wonder if it would be more efficient to signal this thread that some work has been queued rather than poll every 50 milliseconds.

Reply:
I will create a follow up task for this.

Reply:
I'm surprised we're polling, if the task queue [supports work stealing](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskscheduler?view=netframework-4.8#work-stealing). Do we need to do anything special to make the scheduler aware of this?

Reply:
Not sure I follow your comment :) 

The root cause of our timeouts is work stealing. 
These tasks are being scheduled from outside of the threadpool which means they end up on the global queue. 

This is just our implementation of a simple thread that executes async tasks.